### DISCUSSION TITLE
Status of KFP V2

### DISCUSSION THREAD

**Discussion by AndersBennedsgaard** [kubeflow/pipelines#10863] [General]
Hello :smiley: 

I was curious: what is the status of running a pure V2 instance of pipelines? Has it been tested and is used in production?

https://github.com/kubeflow/pipelines/issues/10402 and other issues makes it seem as if it should be possible now, but I still see a lot of references to code from `apiv1beta1` in the API server, several CRDs still use `v1beta1` such as the `ScheduledWorkflow` (which might be intentional since it doesn't have to be versioned the same as the rest of the project, and so on.

Personally, I've hit the issue that the API Server requires setting the `ML_PIPELINE_VISUALIZATIONSERVER_SERVICE_HOST` and `ML_PIPELINE_VISUALIZATIONSERVER_SERVICE_PORT` environment variables, even though the visualization server isn't used in V2 (I've removed the visualization server from my cluster, and most things seems to be running smoothly). Setting bogus values for these environment variables makes the API server run again. The code for that can be seen here:

https://github.com/kubeflow/pipelines/blob/45f3457d4206d1793ebc4ab60f188fe2a17db4eb/backend/src/apiserver/main.go#L150-L156

I've also hit an issue with the central dashboard erroring out with the pipelines card on the front page (see https://github.com/kubeflow/kubeflow/issues/7584)

**Comment by rimolive** [comment]
We have users running KFPv2 for a while, so I'd assume they are already running in production. The need to set the viz server  env vars and the references to `apiv2beta1` are for backward compatibility purposes and we plan to remove in future releases.

**Reply by AndersBennedsgaard** [reply]
Wouldn't it make sense to cover V1-specific code in an `if backwards_compatible {}` block, so the API server and more wouldn't have to have a large amount of redundant code running? Of course `backwards_compatible` could be true by default.

I'm specifically thinking about stuff like this which spawns multiple goroutines:

https://github.com/kubeflow/pipelines/blob/45f3457d4206d1793ebc4ab60f188fe2a17db4eb/backend/src/apiserver/main.go#L143-L157
